react/components/EXPERIMENTAL_Table/Toolbar/PopoverMenu.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {
+  useRef,
+  useState,
+  FC,
+  useLayoutEffect,
+  useCallback,
+} from 'react'
+
+import Button from '../../Button/index.js'
+
+export default function usePopoverMenu() {
+  const [isBoxVisible, setBoxVisible] = useState(false)
+  const buttonRef = useRef(null)
+
+  const handleOutsideClick = (e: Event) =>
+    buttonRef &&
+    buttonRef.current &&
+    e.target instanceof Node &&
+    !buttonRef.current.contains(e.target) &&
+    setBoxVisible(false)
+
+  useLayoutEffect(() => {
+    isBoxVisible && document.addEventListener('mousedown', handleOutsideClick)
```

Comment:
I think that when you don't need to return a value, you can use an `if` statement instead of `condition && method()`. It makes more sense when it's an attribution or it's inside the render function.